File: src/java/magma/compile/rule/split/locate/LastLocator.java
package magma.compile.rule.split.locate;

import java.util.Optional;
import java.util.stream.Stream;

public class LastLocator implements Locator {
    @Override
    public Optional<Integer> locate(String input, String infix) {
        final var index = input.lastIndexOf(infix);
        return index == -1 ? Optional.empty() : Optional.of(index);
    }
}

File: src/c/magma/app/rule/locate/LastLocator.c
#include "../../../../magma/api/stream/Stream.h"
#include "../../../../magma/api/stream/Streams.h"
#include "../../../../java/util/Optional.h"
struct LastLocator(String infix) implements Locator{
	Stream<Integer> locate(String input){
		var index=input.lastIndexOf(infix());
		return index==-1?Streams.empty():Streams.of(index);
	}
	String unwrap(){
		return this.infix;
	}
	int length(){
		return this.infix.length();
	}
}
